lib/fluent/plugin/buffer.rb

```diff
@@ -172,34 +172,46 @@ def metadata(timekey: nil, tag: nil, variables: nil)
       end
 
       # metadata MUST have consistent object_id for each variation
-      # data MUST be Array of serialized events
+      # data MUST be Array of serialized events, or EventStream
       # metadata_and_data MUST be a hash of { metadata => data }
-      def write(metadata_and_data, bulk: false, enqueue: false)
+      def write(metadata_and_data, format: nil, size: nil, enqueue: false)
         return if metadata_and_data.size < 1
         raise BufferOverflowError, "buffer space has too many data" unless storable?
```

Comment:
Hmm... let me clarify. When BufferOverflow is thrown if "incoming data + exising staged and queued size" is larger than buffer limit size?
From the code, BufferOverflowError is thrown at only this point.